builtin/mainmenu/tab_local.lua

```diff
@@ -107,14 +111,26 @@ if enable_gamebar then
 		btnbar:add_button("game_open_cdb", "", plus_image, fgettext("Install games from ContentDB"))
 	end
 else
+	-- Currently chosen game in gamebar: no gamebar -> no "current" game
 	function current_game()
 		return nil
 	end
+
+	-- Game of currently-selected world, used for disabling settings
+	function selected_game()
+		local list = menudata.worldlist:get_list()
+		local index = filterlist.get_current_index(menudata.worldlist,
+			tonumber(core.settings:get("mainmenu_last_selected_world")))
+		local world = list and index and list[index]
+		local gameid = world and world.gameid
+		print(dump({world = world, gameid = gameid}))
+		return gameid and pkgmgr.find_by_gameid(gameid)
+	end
 end
 
 local function get_disabled_settings(game)
```

Comment:
what's the point of this argument if you have `selected_game`?

Reply:
I'd prefer for the content of selected_game to be moved to get_formspec, and this call to selected_game below removed. That way, both invocations to get_disabled_settings will pass the game in the same way. There's no need for it to change based on whether the gamebar is available